buildscripts/semantic-convention/templates/registry/weaver.yaml

```diff
@@ -0,0 +1,155 @@
+
+  # excluded attributes will be commented out in the generated code
+  # this behavior is fully controlled by jinja templates
+  excluded_attributes: ["messaging.client_id"]
+
+  stable_package_name: opentelemetry.semconv
+
+  # "https://opentelemetry.io/schemas/<SEMCONV_VERSION_TAG>"
+  # this is provided from the command line
+  schema_url: ""
+
+templates:
+  - pattern: schema_url-h.j2
+    filter: .
+    application_mode: single
+  - pattern: semantic_attributes-h.j2
+    filter: >
+      semconv_grouped_attributes({
+        "exclude_root_namespace": $excluded_namespaces,
+        "exclude_stability": if $filter == "any" then [] else ["experimental"] end,
+      })
+      | map({
+          root_namespace: .root_namespace,
+          attributes: .attributes,
+          output: $output,
+          stable_package_name: $stable_package_name + ".attributes",
+          filter: $filter,
+          excluded_attributes: $excluded_attributes[]
+      })
+    application_mode: each
+  - pattern: semantic_metrics-h.j2
+    filter: >
+      semconv_grouped_metrics({
+        "exclude_root_namespace": $excluded_namespaces,
+        "exclude_stability": if $filter == "any" then [] else ["experimental"] end,
+      })
+      | map({
+        root_namespace: .root_namespace,
+        metrics: .metrics,
+        output: $output,
+        stable_package_name: $stable_package_name + ".metrics",
+        filter: $filter
+      })
+    application_mode: each
+
+text_maps:
+  cpp_types:
+    int: int
+    double: double
+    boolean: bool
+    string: const char *
+    string[]: const char *[]
+    template[string]: const char *
+    template[string[]]: const char *[]
+
+  cpp_sync_support:
+    counter: "true"
+    histogram: "true"
+    updowncounter: "true"
+    gauge: "false"
+
+  cpp_sync_instrument_to_int64_type:
+    counter: metrics::Counter<uint64_t>
+    histogram: metrics::Histogram<uint64_t>
+    updowncounter: metrics::UpDownCounter<int64_t>
+    # Not supported yet
+    gauge: metrics::Gauge<uint64_t>
+
+  cpp_sync_instrument_to_double_type:
+    counter: metrics::Counter<double>
+    histogram: metrics::Histogram<double>
+    updowncounter: metrics::UpDownCounter<double>
+    # Not supported yet
+    gauge: metrics::Gauge<double>
+
+  cpp_sync_instrument_to_int64_factory:
+    counter: CreateUInt64Counter
+    histogram: CreateUInt64Histogram
+    updowncounter: CreateInt64UpDownCounter
+    # Not supported yet
+    gauge: CreateUInt64Gauge
+
+  cpp_sync_instrument_to_double_factory:
+    counter: CreateDoubleCounter
+    histogram: CreateDoubleHistogram
+    updowncounter: CreateDoubleUpDownCounter
+    # Not supported yet
+    gauge: CreateDoubleGauge
+
+  cpp_async_support:
+    counter: "true"
+    histogram: "false"
+    updowncounter: "true"
+    gauge: "true"
+
+  cpp_async_instrument_to_type:
+    counter: metrics::ObservableInstrument
+    # Not supported yet
```

Comment:
I think async historgram is not part of the specs - we shouldn't generate any code for it (even under `#if 0`)

Reply:
For async histogram, revised the generation to be silent.

While at it, revised sync gauge to emit code for OPENTELEMETRY_ABI_VERSION_NO >= 2,
since it is now implemented.